File: config/genconfig.m
%{
% generate the config for the current simulation
%}
function genconfig(filename, sheetname, casename)
    % Specify the file path and sheet number
    filename = './config/' + filename + ".xlsx";
    
    % Read the data from the Excel file
    sheet_names = sheetnames(filename);
    sheet_num = length(sheet_names);
    
    is_sheet = false;

    for i = 1:sheet_num
        sheet_name = sheet_names(i);
        if sheet_name == sheetname
            is_sheet = true;
            sheet_content = readtable(filename, 'Sheet', i, "VariableNamingRule", 'preserve', 'Range', 'A1:Z100');
            break;
        end

        
    end

    if ~is_sheet
        error("[%s] does not exist!!!", sheetname);
    end
end

File: config/genconfig.py
import inspect
import os
import pandas as pd


def genconfig(filename, sheetname, casename):
    # callers globel
    cgs = inspect.currentframe().f_back.f_globals
    
    # Specify the file path and sheet number
    filename = filename + ".xlsx";
    
    filename = os.path.abspath(os.path.join(os.path.dirname(__file__), filename))
       
    sheet = pd.read_excel(io=filename, 
                          sheet_name=sheetname, 
                          engine='openpyxl',
                          dtype=str)
    for i, row in sheet.iterrows():
        props_name = row.iloc[0]
        prop = row[casename]
        
        cgs[props_name] = eval(prop)